src/app-check/app-check-api-client-internal.ts

```diff
@@ -86,6 +87,31 @@ export class AppCheckApiClient {
       });
   }
 
+  public verifyOneTimeProtection(token: string): Promise<boolean> {
+    if (!validator.isNonEmptyString(token)) {
+      throw new FirebaseAppCheckError(
+        'invalid-argument',
+        '`token` must be a non-empty string.');
+    }
+    return this.getVerifyTokenUrl()
+      .then((url) => {
+        const request: HttpRequestConfig = {
+          method: 'POST',
+          url,
+          headers: FIREBASE_APP_CHECK_CONFIG_HEADERS,
+          data: { token }
+        };
+        return this.httpClient.send(request);
+      })
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      .then((resp) => {
+        return true;
+      })
+      .catch((err) => {
+        throw this.toFirebaseError(err);
```

Comment:
Ah good point! Since we check the validity of the token prior to making the call to BE, #1 should be addressed already. How do we check #2, is it a custom claim encoded in the JWT?

Reply:
For (2), yes, it is actually a private claim named `provider` in the JWT, but now that I think about this more, I think a more robust method for detecting whether the token supports Replay Protection is to see whether its `jti` claim is populated with a nonempty string rather than checking the `provider` claim against a hardcoded list of allowed providers.